sound/soc/sof/intel/hda-stream.c

```diff
@@ -539,26 +557,29 @@ int hda_dsp_stream_init(struct snd_sof_dev *sdev)
 	}
 
 	/* mem alloc for the position buffer */
-	ret = snd_dma_alloc_pages(SNDRV_DMA_TYPE_DEV, &pci->dev, 8,
-				  &hdev->posbuffer);
+	/* TODO: check postion buffer update */
+	ret = snd_dma_alloc_pages(SNDRV_DMA_TYPE_DEV, &pci->dev, 8 * num_total,
```

Comment:
Where does 8 come from ?

Reply:
8 Bytes is the size of DMA posistion for each stream defined in HDA spec, let me change it to macro also.